extensions/spark/kyuubi-spark-connector-hive/src/test/scala/org/apache/kyuubi/spark/connector/hive/KyuubiHiveTest.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.spark.connector.hive
+
+import java.io.File
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.SparkSession
+
+import org.apache.kyuubi.{KyuubiFunSuite, Utils}
+
+abstract class KyuubiHiveTest extends KyuubiFunSuite {
+
+  private var spark: SparkSession = _
+
+  def withSparkSession[T](conf: Map[String, String] = Map.empty[String, String])(
+      f: SparkSession => T): T = {
+    val namePrefix = s"${this.getClass.getSimpleName}-${Thread.currentThread().getName}"
+    val tmp = Utils.createTempDir(namePrefix = namePrefix).toFile.getAbsolutePath
+
+    val sparkConf = new SparkConf()
+      .setMaster("local[*]")
+      .set("spark.ui.enabled", "false")
+      .set("spark.sql.catalogImplementation", "hive")
+      .set("spark.sql.catalog.v2hive", classOf[HiveTableCatalog].getName)
+      .set(
+        "javax.jdo.option.ConnectionURL",
+        s"jdbc:derby:;databaseName=$tmp/metastore_db;create=true")
```

Comment:
I'm wondering if the in-memory mode is applicable.
`https://db.apache.org/derby/docs/10.16/devguide/cdevdvlpinmemdb.html`